File: internal/indexer/testdata/java/DefsA.java
package com.example;

import java.util.List;

/**
 * AppConfig holds application configuration.
 * Supports multiple environments.
 */
public class AppConfig {
    private String host;
    private int port;

    /**
     * Create a new AppConfig.
     */
    public AppConfig(String host, int port) {
        this.host = host;
        this.port = port;
    }

    /**
     * Returns the host.
     */
    public String getHost() {
        return host;
    }

    /**
     * Returns the port.
     */
    public int getPort() {
        return port;
    }

    public void validate() {
        if (host == null || host.isEmpty()) {
            throw new IllegalArgumentException("host is required");
        }
        if (port <= 0) {
            throw new IllegalArgumentException("port must be positive");
        }
    }
}

/**
 * Formatter interface for output formatting.
 */
interface AppFormatter {
    String format(String data);
    void reset();
}

@Deprecated
enum AppStatus {
    ACTIVE,
    INACTIVE,
    MAINTENANCE
}

File: internal/indexer/testdata/rust/defs_a.rs
use std::fmt;

/// AppConfig holds application configuration.
/// Supports multiple environments.
pub struct AppConfig {
    pub host: String,
    pub port: u16,
    pub log_level: String,
}

impl AppConfig {
    /// Creates a new AppConfig with defaults.
    pub fn new(host: String) -> Self {
        AppConfig {
            host,
            port: 8080,
            log_level: "info".to_string(),
        }
    }

    /// Validates the configuration.
    pub fn validate(&self) -> Result<(), String> {
        if self.host.is_empty() {
            return Err("host is required".to_string());
        }
        if self.port == 0 {
            return Err("port must be positive".to_string());
        }
        Ok(())
    }
}

/// Formatter trait for output formatting.
pub trait AppFormatter {
    fn format(&self, data: &str) -> String;
    fn reset(&mut self);
}

/// Application status.
#[derive(Debug, Clone)]
pub enum AppStatus {
    Active,
    Inactive,
    Maintenance,
}

/// Default port constant.
const DEFAULT_PORT: u16 = 8080;

/// Application name.
type AppName = String;
